### PR TITLE
Release 1.12.0

### PR DESCRIPTION
**Pull request by kunaltyagi**
- [x] Go through all PR for the milestone
- [x] Create a pre-release via the release pipeline
- [x] (If needed) Revert whatever unintentional API/ABI breakage that might have occurred in the release branch
- [ ] Fix issues exposed via pre-release
- [ ] Update Changelog. Generate it via `.dev/scripts/generate_changelog.py`
- [x] Bump version via `.dev/scripts/bump_release.bash 1.12.0`

### CODE REVIEW

**Comment by kunaltyagi** [comment]
Ignore the release CI failure. It fails if it's not run manually

File: CHANGES.md [hunk 1]
```diff
@@ -1,5 +1,227 @@
 # ChangeList
 
+## = 1.12.0 (02.06.2021) =
```
Comment (jspricke) [inline CHANGES.md:3]: nit: date is in the past ;).

**Review by jspricke — APPROVED** [review]

**Comment by kunaltyagi** [comment]
Closing this PR. I think changelog should go in a separate PR for the pre-release

**Comment by kunaltyagi** [comment]
That'd allow the pre-release note to have a link to the CHANGES.md in it

**Pull request closed without merging**